concourse/scripts/tag_release_test.go

```diff
@@ -0,0 +1,219 @@
+package scripts_test
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	"github.com/onsi/gomega/gexec"
+)
+
+var _ = Describe("TagRelease", func() {
+
+	var (
+		workingDirectory string
+		originRepoPath   string
+		baseCmd          *exec.Cmd
+	)
+
+	BeforeEach(func() {
+		gitVersionCommand := exec.Command("git", "--version")
+
+		session, err := gexec.Start(gitVersionCommand, GinkgoWriter, GinkgoWriter)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(session).Should(gexec.Exit(0))
+		Expect(session.Out.Contents()).To(MatchRegexp(".* [2-9]\\.[0-9]+\\.[0-9]+.*"), "Expected git version 2.0.0 or bigger")
+		Expect(session.Err.Contents()).To(BeEmpty())
+	})
+
+	BeforeEach(func() {
+		workingDirectory, baseCmd = prepareWorkingDirectory(
+			"./tag_release.sh",
+		)
+
+		prepareSshKeys(baseCmd)
+		configGitEmailAndName(baseCmd)
+
+		originRepoPath = setupClonedRepository(baseCmd)
+		baseCmd.Env = append(baseCmd.Env, fmt.Sprintf("GIT_REPO_URL=%s", originRepoPath))
+	})
+
+	AfterEach(func() {
+		cleanUpWorkingDirectory(workingDirectory)
+	})
+
+	Context("when there are multiple 'previous-*' tags to be promoted", func() {
+		BeforeEach(func() {
+			runBashScript(baseCmd, `
+        cd origin_repo
```

Comment:
The indenting of these inline bash scripts seem a little off, making this hard to read.

In the original the code of the bash scripts are indented at one level lower than the surrounding code, so that would make this:

```
                BeforeEach(func() {
                        runBashScript(baseCmd, `
                                cd origin_repo

                                git commit --allow-empty -m "First commit"

                                git tag previous-0.0.1
                                git tag previous-0.0.2

                                git commit --allow-empty -m "Second commit"
                                git tag previous-0.0.3

                                # cannot push to repo while its checked out to a branch
                                git checkout refs/heads/master
                        `)
                })
```